File: java/src/main/java/problems1_100/RemoveDuplicatesFromSortedList.java
package problems1_100;

import concepts.SinglyListNode;

enum RemoveDuplicatesFromSortedList {
    DEFAULT {
        @Override
        SinglyListNode deleteDuplicates(SinglyListNode head) {
            SinglyListNode current = head;
            while (current != null && current.next != null) {
                if (current.val == current.next.val) {
                    current.next = current.next.next;
                } else {
                    current = current.next;
                }
            }
            return head;
        }
    };

    abstract SinglyListNode deleteDuplicates(SinglyListNode head);
}

File: kotlin/src/main/kotlin/problems1_100/RemoveDuplicatesFromSortedList.kt
package problems1_100

import concepts.SinglyListNode

enum class RemoveDuplicatesFromSortedList {
    DEFAULT {
        override fun deleteDuplicates(head: SinglyListNode?): SinglyListNode? {
            var current = head
            while (current?.next != null) {
                if (current.`val` == current.next!!.`val`) {
                    current.next = current.next!!.next
                } else {
                    current = current.next
                }
            }
            return head
        }
    },
    ;

    abstract fun deleteDuplicates(head: SinglyListNode?): SinglyListNode?
}
